amethyst_core/src/named.rs

```diff
@@ -136,12 +136,11 @@ impl<'a> WithNamed for EntityBuilder<'a> {
     where
         S: Into<Cow<'static, str>>,
     {
-        // Unwrap: The only way this can fail is if the entity is invalid and this is used while creating the entity.
         self.world
             .system_data::<(WriteStorage<'a, Named>,)>()
             .0
             .insert(self.entity, Named::new(name))
-            .unwrap();
+            .expect("entity should always be valid when just created");
```

Comment:
The unwrap was actually fine and had an explanation. There wasn't much reason for the change.

Reply:
True I figured for concistency and better reports if it will ever fail because of a bug. Should I not do this in other places + revert this one?

Reply:
You could change the message to
"unreachable: Entities should always be valid when just created"
?

:)

Reply:
I think replacing all unwraps with expects makes it much better.
Last time, when we had that camera crash, it took too long to discover where the bug was from without an expect telling us.

Reply:
Sure! Should I then do the same if I encounter other unwraps like this?

Reply:
Yep, seems good to me.